httpx/exceptions.py

```diff
@@ -1,7 +1,20 @@
+class HttpxError(Exception):
+    """
+    Base class for Httpx exception
+    """
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Could we use `request: BaseRequest = None, response: BaseResponse = None` here?
Would be better to have the tighter checking if possible.

Reply:
I'll probably leave the *args as it allows to pass message to the exception. And I think we have circular import if we try to import BaseRequest and BaseResponse in exceptions, I'll leave them untyped for now if it's OK with you

Reply:
Actually there's an entry in the doc for this kind of issue: https://mypy.readthedocs.io/en/latest/common_issues.html#import-cycles I'll take a look this evening to fix this.

Reply:
Ah interesting yeah - that's worth a shot.